src/core.js

```diff
@@ -101,52 +124,105 @@ async function pruneNonCriticalCssLauncher ({
+              if (pageLoadSkipTimeout) {
+                debuglog('pageLoadSkipTimeout injected on dom creation')
+                // This is crucial for the evaluate to work. If this is not set we got an error:
+                // ERROR Error: Protocol error (Runtime.callFunctionOn): Cannot find context with specified id undefined
+                // Maybe this should be evaluated on low power machines if they need 100ms
+                await page.waitFor(500)
+                page
+                  .evaluate(pageLoadSkipTimeoutFunc, {
+                    pageLoadSkipTimeout
+                  })
+                  .then(message => {
+                    if (!_hasLoaded) {
+                      // when pageLoadSkipTimeout is reached after dom request
+                      // don't resolve when page already loaded completely
+                      debuglog(
+                        'pageLoadSkipTimeout - page load waiting ABORTED after ' +
+                          pageLoadSkipTimeout / 1000 +
+                          's. '
+                      )
+                      return resolve('pageLoadSkipTimeout')
+                    }
+                  })
+                  .catch(err => {
+                    if (!err.message.includes('Target closed')) {
+                      debuglog('page.evaluate - ERROR', err)
+                      _hasError = err
+                      return reject(err)
+                    }
+                  })
+              }
+            }
+          })
         }
       })
 
       debuglog('page load start')
       // set a higher number than the timeout option, in order to make
       // puppeteer’s timeout _never_ happen
-      let waitingForPageLoad = true
-      const loadPagePromise = page.goto(url, { timeout: timeout + 1000 })
-      if (pageLoadSkipTimeout) {
-        await Promise.race([
-          loadPagePromise,
-          new Promise(resolve => {
-            // instead we manually _abort_ page load after X time,
-            // in order to deal with spammy pages that keep sending non-critical requests
-            // (tracking etc), which would otherwise never load.
-            // With JS disabled it just shouldn't take that many seconds to load what's needed
-            // for critical viewport.
-            setTimeout(() => {
-              if (waitingForPageLoad) {
-                debuglog(
-                  'page load waiting ABORTED after ' +
-                    pageLoadSkipTimeout / 1000 +
-                    's. '
-                )
-                resolve()
-              }
-            }, pageLoadSkipTimeout)
+      const loadPageResponse = new Promise((resolve, reject) => {
+        page
+          .goto(url, {
+            timeout: timeout + 1000,
+            waitUntil: 'networkidle'
+          })
+          .then(response => {
+            // Reject exits the Promise.all call immediately and ensures that the pageload is always higher prio than timout
```

Comment:
I think you are mixing up Promise.all and Promise.race here - this comment is not accurate. (this fulfills the promise, hopefully winning the race. Would not do what you want if it was promise.all below).

Reply:
It is a relic of my try to replace race with all. Just a wrong comment.